src/erlcloud_kms.erl

```diff
@@ -0,0 +1,1060 @@
+         put_key_policy/3, put_key_policy/4,
+         re_encrypt/2, re_encrypt/3, re_encrypt/4,
+         retire_grant/1, retire_grant/2,
+         revoke_grant/2, revoke_grant/3,
+         update_alias/2, update_alias/3,
+         update_key_description/2, update_key_description/3]).
+
+
+%%%------------------------------------------------------------------------------
+%%% Shared types
+%%%------------------------------------------------------------------------------
+-type pagination_opts() :: [pagination_opt()].
+-type pagination_opt() :: {limit, non_neg_integer()} | {marker, string()}.
+
+
+%%%------------------------------------------------------------------------------
+%%% Library initialization.
+%%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{
+       access_key_id=AccessKeyID,
+       secret_access_key=SecretAccessKey
+      }.
+
+-spec new(string(), string(), string()) -> aws_config().
+
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{
+       access_key_id=AccessKeyID,
+       secret_access_key=SecretAccessKey,
+       s3_host=Host
+      }.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{
+       access_key_id=AccessKeyID,
+       secret_access_key=SecretAccessKey,
+       s3_host=Host,
+       s3_port=Port
+      }.
+
+-spec configure(string(), string()) -> ok.
+
+configure(AccessKeyID, SecretAccessKey) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey)),
+    ok.
+
+-spec configure(string(), string(), string()) -> ok.
+
+configure(AccessKeyID, SecretAccessKey, Host) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer()) -> ok.
+
+configure(AccessKeyID, SecretAccessKey, Host, Port) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port)),
+    ok.
+
+default_config() ->
+    erlcloud_aws:default_config().
+
+%%%------------------------------------------------------------------------------
+%%% CreateAlias
+%%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% KMS API:
+%% [http://docs.aws.amazon.com/kms/latest/APIReference/API_CreateAlias.html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+-spec create_alias/2 ::
+          (AliasName :: string(),
+           TargetKeyId :: string()) ->
+          any().
+create_alias(AliasName, TargetKeyId) ->
+    create_alias(AliasName, TargetKeyId, default_config()).
+
+
+-spec create_alias/3 ::
+          (AliasName :: string(),
+           TargetKeyId :: string(),
+           Config :: aws_config()) ->
+          any().
```

Comment:
i doubt `any()` spec has any value `erlcloud_aws:request` has pretty limited set of returns